### PR TITLE
CSP-869_Employer Onboarding MultiRegion Check Your Answers

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**

### CODE REVIEW

File: src/SFA.DAS.Employer.Aan.Web.UnitTests/Controllers/Onboarding/AreasToEngageLocallyControllerTests/AreasToEngageLocallyControllerPostTests.cs [hunk 1]
```diff
@@ -35,15 +35,19 @@ public class AreasToEngageLocallyControllerPostTests
         sessionServiceMock.Verify(s => s.Set(It.Is<OnboardingSessionModel>(m => m.Regions.Find(x => x.Id == submitmodel.SelectedAreaToEngageLocallyId)!.IsConfirmed)));
     }
 
-    [Test, MoqAutoData]
-    public void Post_RedirectsTo_JoinTheNetworkPage(
+    [MoqInlineAutoData(false, RouteNames.Onboarding.JoinTheNetwork)]
```
Comment (MarkFCain) [inline src/SFA.DAS.Employer.Aan.Web.UnitTests/Controllers/Onboarding/AreasToEngageLocallyControllerTests/AreasToEngageLocallyControllerPostTests.cs:38]: These look like they're missing the Test attribute

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.Employer.Aan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/WhenGetIsInvoked/AndSessionModelIsNotPopulated.cs [hunk 2]
```diff
@@ -44,9 +49,9 @@ public void ThenReturnsViewResults()
     }
 
     [Test]
-    public void ThenSetsEmployerAccountIdInTheViewModel()
+    public void ThenSetsEmployerAccountIdToNullInTheViewModel()
     {
-        _viewModel.EmployerAccountId.Should().Be(_employerAccountId);
+        _viewModel.EmployerAccountId.Should().BeNull();
```
Comment (DevenKShah) [inline src/SFA.DAS.Employer.Aan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/WhenGetIsInvoked/AndSessionModelIsNotPopulated.cs:54]: This should never be null

File: src/SFA.DAS.Employer.Aan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/WhenGetIsInvoked/AndSessionModelIsNotPopulated.cs
Comment (DevenKShah) [inline src/SFA.DAS.Employer.Aan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/WhenGetIsInvoked/AndSessionModelIsNotPopulated.cs:1]: Rename this to `AndUserHasSkippedJourney`

File: src/SFA.DAS.Employer.Aan.Web/Controllers/Onboarding/JoinTheNetworkController.cs [hunk 3]
```diff
@@ -63,33 +63,33 @@ public IActionResult Post(JoinTheNetworkSubmitModel submitModel)
...
             Support = new List<SelectProfileModel>(sessionModel.ProfileData.Where(x => x.Category == Category.Support).OrderBy(x => x.Ordering).Select(x => (SelectProfileModel)x))
         };
     }
 
-    private string GetCorrectBackLink(OnboardingSessionModel sessionModel)
+    private string GetCorrectBackLink(OnboardingSessionModel sessionModel, string employerAccountId)
     {
         var noOfRegionsSelected = sessionModel.Regions.Count(x => x.IsSelected);
 
         if (noOfRegionsSelected == 1)
```
[the remark this answers is not in the document]
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.Employer.Aan.Web/Controllers/Onboarding/JoinTheNetworkController.cs:81]: Can't implement pattern matching here due to multiple deciding variable for user flow

File: src/SFA.DAS.Employer.Aan.Web/Models/Onboarding/CheckYourAnswersViewModel.cs [hunk 4]
```diff
@@ -37,6 +46,10 @@ public CheckYourAnswersViewModel(IUrlHelper url, OnboardingSessionModel sessionM
         {
             regions.Add($"Locally prefers {locallyPreferredRegion.Area}");
         }
+        if (sessionModel.IsLocalOrganisation.HasValue && !sessionModel.IsLocalOrganisation.GetValueOrDefault())
```
[the remark this answers is not in the document]
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.Employer.Aan.Web/Models/Onboarding/CheckYourAnswersViewModel.cs:49]: When the nullable field sessionModel.IsLocalOrganisation contains null then GetValueOrDefault() return false which we do not want, hence used this way

**Comment by YuvrajPatil-DFE** [comment]
Closing this PR as it was merged into CSP-636

**Pull request closed without merging**